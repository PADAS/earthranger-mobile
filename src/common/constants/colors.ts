const COLORS_LIGHT = {
  G0_black: '#000000',
  G1_off_black: '#333333',
  G2_secondaryMediumGray: '#63666A',
  G2_5_mobileSecondaryGray: '#767676',
  G3_secondaryMediumLightGray: '#888B8D',
  G4_secondaryLightGray: '#B1B3B3',
  G5_LightGreyLines: '#DDDDDD',
  G6_LightGreyButton: '#F3F3F3',
  G7_veryLightGrey: '#F9F9F9',
  white: '#FFFFFF',
  transparent: '#FFFFFF00',
  blackTransparent88: '#00000088',
  brightBlue: '#0056C7',
  indigo: '#3E35A3',
  magenta: '#B62879',
  darkTeal: '#107283',
  blueLight: '#DDE8F8',
  green: '#006842',
  red: '#D0031B',
  redLight: '#FDF2F4',
  amber: '#E76826',
  erTeal: '#008586',
  blue2: '#F4F7FD',
};

const COLORS_DARK = {
  G0_black: '#FFFFFF',
  G2_secondaryMediumGray: '#B1B3B3',
  G2_5_mobileSecondaryGray: '#888B8D',
  G3_secondaryMediumLightGray: '#888B8D',
  G4_secondaryLightGray: '#333333',
  G5_LightGreyLines: '#444444',
  G6_LightGreyButton: '#F3F3F3',
  G7_veryLightGrey: '#111111',
  white: '#222222',
  transparent: '#FFFFFF00',
  blackTransparent88: '#00000088',
  brightBlue: '#76B0FF',
  indigo: '#867EDE',
  magenta: '#E76EB3',
  darkTeal: '#62BFD0',
  blueLight: '#F4F7FD',
  green: '#83AF52',
  red: '#F06374',
  amber: '#FD9964',
  erTeal: '#008586',
  blue2: '#F4F7FD',
};

export { COLORS_LIGHT, COLORS_DARK };
