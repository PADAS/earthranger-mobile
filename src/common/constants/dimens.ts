export const BOTTOM_TAB_BAR_HEIGHT = 49;
export const PATROL_DETAILS_TITLE_HEIGHT = 88;
export const PATROL_DETAILS_CONTENT_HEIGHT = 310;
export const TRACKING_STATUS_TITLE_HEIGHT = 88;
export const TRACKING_STATUS_INLINE_BUTTONS_HEIGHT = 160;
export const TRACKING_STATUS_STACKED_BUTTONS_HEIGHT = 210;
export const MAP_OVERLAY_BUTTON_SIZE = 49;
export const MAP_OVERLAY_BUTTON_BOTTOM_MARGIN = 16;
export const SUBJECT_DETAILS_CONTENT_HEIGHT = 150;

const margin = {
  small: 8,
  standard: 16,
};

export { margin };
