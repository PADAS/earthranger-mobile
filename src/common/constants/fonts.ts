type TFontWeightValue = 'normal' | 'bold' | '100' | '200' | '300' | '400' | '500' | '600' | '700' | '800' | '900';
type TFontWeightName = 'normal' | 'medium' | 'semiBold' | 'bold';
type TFontSizeName = 'medium' | 'standard' | 'large' | 'larger' | 'huge' | 'massive';
type TLineHeightName = 'medium' | 'standard' | 'large' | 'huge' | 'massive';

const MAX_FONT_SIZE_MULTIPLIER = 1.1;

const fontSize: Record<TFontSizeName, number> = {
  medium: 12,
  standard: 14,
  large: 16,
  larger: 17,
  huge: 18,
  massive: 25,
};

const lineHeight: Record<TLineHeightName, number> = {
  medium: 14,
  standard: 16,
  large: 18,
  huge: 21,
  massive: 25,
};

const fontWeight: Record<TFontWeightName, TFontWeightValue> = {
  normal: '400',
  medium: '500',
  semiBold: '600',
  bold: '700',
};

export {
  fontSize,
  lineHeight,
  fontWeight,
  MAX_FONT_SIZE_MULTIPLIER,
};
