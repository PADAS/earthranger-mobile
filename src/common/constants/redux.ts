// Internal Dependencies
import { InitialState } from '../types/redux';

// Initial State
export const initialState: InitialState = {
  session: {
    accessToken: '',
    refreshToken: '',
    error: null,
    sessionError: null,
  },
  reports: {
    noteId: -1,
    textNote: '',
  },
};

// Action Types
export const USER_LOGIN = 'USER_LOGIN';
export const USER_LOGIN_ERROR = 'USER_LOGIN_ERROR';
export const GET_SESSION = 'GET_SESSION';
export const GET_SESSION_ERROR = 'GET_SESSION_ERROR';
export const CREATE_NOTE = 'CREATE_NOTE';
export const CLEAR_NOTE_DATA = 'CLEAR_NOTE_DATA';
export const CLEAR_SESSION = 'CLEAR_SESSION';
