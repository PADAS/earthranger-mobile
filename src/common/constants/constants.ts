// External Dependencies
import { Platform } from 'react-native';

// Internal Dependencies
import { COLORS_LIGHT } from './colors';

// Data
export const DATABASE_FILE_NAME = 'earthranger.db';
export const TRANSISTOR_LOCATION_MANAGER = 'transistor_location_manager';

// Adjustment values for small screens iphone 5, 6, 7, SE and Android Nexus One
export const SCREEN_FIT_700_TO_600 = 0.04;
export const SCREEN_FIT_LESS_THAN_600 = 0.4;
export const DATE_FORMAT_HHMM_DD_MMM_YYYY = 'HH:mm DD MMM YYYY';
export const DATE_FORMAT_YYYY_MM_DD_HH_MM = 'YYYY MMM DD, HH:mm';
export const ALERT_BUTTON_BACKGROUND_COLOR_BLUE = COLORS_LIGHT.brightBlue;
export const ALERT_BUTTON_BACKGROUND_COLOR_RED = COLORS_LIGHT.red;

// Network
export const NETWORK_ERROR = 'Network Error';

// Platform
export const IS_ANDROID = Platform.OS === 'android';
export const IS_IOS = Platform.OS === 'ios';

// Key-Value Storage - Keys
export const ACTIVE_PATROL_KEY = 'activePatrolKey';
export const LAST_SYNC_PATROLS_TIME_KEY = 'lastSyncPatrolsTimeKey';
export const LAST_SYNC_REPORTS_TIME_KEY = 'lastSyncReportsTimeKey';
export const LAST_SYNC_LOCATION_TIME_KEY = 'lastSyncLocationTimeKey';
export const LOCATION_STATUS_KEY = 'locationStatusKey';
export const MERGE_CATEGORIES_KEY = 'mergeCategoriesKey';
export const PATROL_STATUS_KEY = 'patrolStatusKey';
export const PROMINENT_DISCLOSURE_KEY = 'prominentDisclosureKey';
export const REMEMBER_ME_CHECKBOX_KEY = 'rememberMeCheckBoxKey';
export const SHOW_USER_TRAILS_ENABLED_KEY = 'showUserTrailsEnabledKey';
export const SYNC_TRACKS_MANUALLY = 'syncTracksManually';
export const STORE_LANGUAGE_KEY = 'language';
export const UPDATE_PATROL_TYPES_UI = 'updatePatrolTypesUI';
export const IS_DEVICE_TRACKING = 'isDeviceTrackingKey';
export const BASEMAP_KEY = 'basemapKey';
export const BASEMAP_TOPO_LOCAL = 'basemapTopoLocal';
export const BASEMAP_STREET_LOCAL = 'basemapStreetLocal';
export const BASEMAP_SATELLITE_LOCAL = 'basemapSatelliteLocal';
export const REPORTS_SUBMITTED_KEY = 'reportsSubmitted';
export const TRACKED_BY_SUBJECT_NAME_KEY = 'trackBySubjectNameKey';
export const TRACKED_BY_SUBJECT_ID_KEY = 'tracedBySubjectIdKey';
export const TRACKED_BY_SUBJECT_STATUS_KEY = 'tracedBySubjectStatusKey';
export const ACTIVE_PATROL_ID_KEY = 'activePatrolIdKey';
export const ACTIVE_USER_HAS_PATROLS_PERMISSION = 'activeUserHasPatrolsPermission';
export const PHOTO_QUALITY_KEY = 'PhotoQualityKey';
export const SAVE_TO_CAMERA_ROLL = 'SaveToCameraRoll';
export const UPLOAD_PHOTOS_WIFI = 'UploadPhotosWifi';
export const PATROL_START_LOCATION = 'PatrolStartLocation';
export const PATROL_EVENT_DETAILS = 'PatrolEventDetails';
export const PATROL_DEFAULT_EVENT_TYPE_VALUE = 'PatrolDefaultEventTypeValue';
export const IS_STATUS_FILTER_DRAFT_SELECTED = 'StatusFilterDraft';
export const IS_STATUS_FILTER_PENDING_SELECTED = 'StatusFilterPending';
export const IS_STATUS_FILTER_SYNCED_SELECTED = 'StatusFilterSynced';
export const IS_STATUS_FILTER_ENABLED = 'IsStatusFilterEnabled';

// Event Emitter Keys
export const REPORTS_SYNCING = 'REPORTS_SYNCING';
export const PATROLS_SYNCING = 'PATROLS_SYNCING';
export const LOCAL_DB_SYNCING = 'LOCAL_DB_SYNCING';
export const COORDINATES_FORMAT_KEY = 'coordinatesFormatKey';
export const START_PATROL_EVENT = 'START_PATROL_EVENT';
export const BOTTOM_SHEET_NAVIGATOR = 'BOTTOM_SHEET_NAVIGATOR';
export const BOTTOM_SHEET_NAVIGATOR_STATUS_INDEX = 'BOTTOM_SHEET_NAVIGATOR_STATUS';

// Experimental Features
export const EXPERIMENTAL_FEATURES_FLAG_KEY = 'experimentalFeaturesGlagKey';
export const SETTINGS_VIEW_DISAPPEAR_KEY = 'settingsViewDisappearKey';

// MMKV Keys - Secure
export const AREA_INSTRUCTIONS_VIEWED = 'areaInstructionsViewed';
export const CURRENT_MENU_SETTINGS_TAB = 'currentMenuSettingsTab';
export const PATROL_TYPE = 'patrolType';
export const PATROL_TYPE_DISPLAY = 'patrolTypeDisplay';
export const AUTH_STATE = 'authState';
export const SESSION_KEY = 'sessionKey';
export const SITE_VALUE_KEY = 'siteValueKey';
export const SUBJECT_ID_KEY = 'subjectIdKey';
export const USER_ID_KEY = 'userIdKey';
export const PARENT_USER_REMOTE_ID_KEY = 'parentUserRemoteIdKey';
export const USER_NAME_KEY = 'userNameKey';
export const USER_SUBJECT_NAME_KEY = 'userSubjectNameKey';
export const ACTIVE_USER_NAME_KEY = 'activeUserNameKey';
export const USER_REMOTE_ID_KEY = 'userRemoteIdKey';
export const DEVICE_UUID = 'deviceUuid';

// Geometry
export const COORDINATES_SCALE = 100000;
export const MAPBOX_MAXIMUM_LATITUDE = 85.0511;
