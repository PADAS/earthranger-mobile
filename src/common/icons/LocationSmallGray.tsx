// External Dependencies
import React from 'react';
import {
  Svg, Path,
} from 'react-native-svg';

// Internal Dependencies
import { COLORS_LIGHT } from '../constants/colors';
import { IconProps } from '../types/types';

const LocationSmallGray = ({ width = '8', height = '8', color = COLORS_LIGHT.G3_secondaryMediumLightGray }: IconProps) => (
  <Svg width={width} height={height} viewBox="0 0 8 8" fill={color}>
    <Path d="M8 0L0 3.34667V3.78222L3.04 4.96L4.21333 8H4.64889L8 0Z" />
  </Svg>
);

export { LocationSmallGray };
