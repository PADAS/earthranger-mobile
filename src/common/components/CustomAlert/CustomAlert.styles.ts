// External Dependencies
import { StyleSheet } from 'react-native';

// Internal Dependencies
import { COLORS_LIGHT } from '../../constants/colors';
import { fontSize, fontWeight, lineHeight } from '../../constants/fonts';

const customAlertStyles = StyleSheet.create({
  modalOverlay: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: COLORS_LIGHT.blackTransparent88,
  },
  modalContainer: {
    width: '80%',
    backgroundColor: COLORS_LIGHT.white,
    borderRadius: 5,
    paddingTop: 22,
    paddingBottom: 22,
    paddingLeft: 15,
    paddingRight: 15,
  },
  header: {
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
  },
  headerIcon: {
    marginRight: 8,
  },
  messageText: {
    color: COLORS_LIGHT.G2_secondaryMediumGray,
    fontSize: 15,
    marginTop: 16,
    fontWeight: '400',
  },
  additionalText: {
    fontWeight: 'bold',
  },
  horizontalView: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginTop: 22,
    alignItems: 'center',
  },
  headerText: {
    flex: 1,
    fontSize: fontSize.huge,
    lineHeight: lineHeight.huge,
    fontWeight: fontWeight.medium,
  },
  negativeButton: {
    flexDirection: 'row',
  },
  negativeButtonText: {
    lineHeight: 18,
    fontSize: 17,
    fontWeight: fontWeight.semiBold,
    color: COLORS_LIGHT.G3_secondaryMediumLightGray,
    textTransform: 'capitalize',
  },
  positiveButton: {
    height: 40,
    alignItems: 'center',
    borderRadius: 5,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    marginLeft: 24,
    maxWidth: '60%',
    minWidth: '40%',
    paddingHorizontal: 16,
  },
  positiveText: {
    fontSize: 17,
    lineHeight: 21,
    marginRight: 9,
    marginLeft: 9,
    fontWeight: fontWeight.semiBold,
    color: COLORS_LIGHT.white,
    textTransform: 'capitalize',
  },
  positiveIcon: {
    marginRight: 6,
    marginLeft: 9,
  },
  progressBar: {
    height: 12,
    marginTop: 16,
    backgroundColor: COLORS_LIGHT.G5_LightGreyLines,
    borderRadius: 5,
  },
});

export { customAlertStyles };
