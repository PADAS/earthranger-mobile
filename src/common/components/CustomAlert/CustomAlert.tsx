// External Dependencies
import React from 'react';
import {
  Modal, View, Pressable,
} from 'react-native';
import { Text } from 'react-native-ui-lib';
import { isEmpty } from 'lodash-es';
import { ProgressBar } from 'react-native-paper';

// Internal Dependencies
import { COLORS_LIGHT } from '../../constants/colors';

// Styles
import { customAlertStyles } from './CustomAlert.styles';

// Interfaces + Types
interface ICustomAlertProps {
  additionalText?: string;
  alertIcon?: JSX.Element;
  alertMessageText?: string;
  alertTitleText?: string;
  alertTitleTextColor?: string;
  displayAlert: boolean;
  negativeButtonIcon?: React.ReactElement;
  negativeButtonText?: string;
  onNegativeButtonPress?: () => void;
  onPositiveButtonPress: () => void;
  positiveButtonBackgroundColor: string;
  positiveButtonIcon?: React.ReactElement;
  positiveButtonText: string;
  progress?: number;
}

/**
 * Custom Alert - Displays an alert with custom title, message, and CTAs.
 *
 * @param ICustomAlertProps Alert properties to customize it.
 * @returns JSX
 */
const CustomAlert = ({
  additionalText,
  alertIcon,
  alertMessageText,
  alertTitleText,
  alertTitleTextColor,
  displayAlert,
  negativeButtonIcon,
  negativeButtonText,
  onNegativeButtonPress,
  onPositiveButtonPress,
  positiveButtonBackgroundColor,
  positiveButtonIcon,
  positiveButtonText,
  progress,
}: ICustomAlertProps) => (
  <Modal visible={displayAlert} transparent animationType="fade">
    <View style={customAlertStyles.modalOverlay}>
      <View style={customAlertStyles.modalContainer}>
        <View style={customAlertStyles.header}>
          {alertIcon ? (
            <View style={customAlertStyles.headerIcon}>
              {alertIcon}
            </View>
          ) : null}
          {!isEmpty(alertTitleText) && (
          <Text
            style={customAlertStyles.headerText}
            color={alertTitleTextColor || COLORS_LIGHT.G0_black}
          >
            {alertTitleText}
          </Text>
          )}
        </View>
        {!isEmpty(alertMessageText) && (
        <Text style={customAlertStyles.messageText}>
          {alertMessageText}
        </Text>
        )}

        {!isEmpty(additionalText) && (
          <Text style={customAlertStyles.additionalText}>{additionalText}</Text>
        )}

        {progress ? (
          <ProgressBar
            style={customAlertStyles.progressBar}
            progress={progress || 0}
            color={COLORS_LIGHT.brightBlue}
          />
        ) : null}
        <View style={customAlertStyles.horizontalView}>
          {!isEmpty(negativeButtonText) && (
            <Pressable
              style={customAlertStyles.negativeButton}
              onPress={onNegativeButtonPress}
              hitSlop={5}
            >
              {negativeButtonIcon && (
                <View style={customAlertStyles.positiveIcon}>
                  {negativeButtonIcon}
                </View>
              )}
              <Text style={customAlertStyles.negativeButtonText}>
                {negativeButtonText}
              </Text>
            </Pressable>
          )}
          {!isEmpty(positiveButtonText) && (
            <Pressable
              style={[
                { backgroundColor: positiveButtonBackgroundColor },
                customAlertStyles.positiveButton,
              ]}
              onPress={onPositiveButtonPress}
            >
              {positiveButtonIcon && (
              <View style={customAlertStyles.positiveIcon}>
                {positiveButtonIcon}
              </View>
              )}
              <Text style={customAlertStyles.positiveText}>{positiveButtonText}</Text>
            </Pressable>
          )}
        </View>
      </View>
    </View>
  </Modal>
);

export { CustomAlert };
