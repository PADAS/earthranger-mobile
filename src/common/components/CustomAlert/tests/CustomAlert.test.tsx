// External Dependencies
import React from 'react';
import { render } from '@testing-library/react-native';

// Internal Dependencies
import { CustomAlert } from '../CustomAlert';
import { ALERT_BUTTON_BACKGROUND_COLOR_BLUE } from '../../../constants/constants';
import { LocationSmallGray } from '../../../icons/LocationSmallGray';

// Mocks
const CUSTOM_ALERT_MOCK = {
  title: 'Alert Title',
  message: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam semper in dolor et egestas.',
  positiveButton: 'Accept',
  negativeButton: 'Reject',
  icon: <LocationSmallGray />,
};

describe('CustomAlert', () => {
  it('matches snapshot', () => {
    const { toJSON } = render(
      <CustomAlert
        displayAlert
        alertTitleText={CUSTOM_ALERT_MOCK.title}
        alertMessageText={CUSTOM_ALERT_MOCK.message}
        positiveButtonText={CUSTOM_ALERT_MOCK.positiveButton}
        negativeButtonText={CUSTOM_ALERT_MOCK.negativeButton}
        onPositiveButtonPress={() => {}}
        onNegativeButtonPress={() => {}}
        positiveButtonBackgroundColor={ALERT_BUTTON_BACKGROUND_COLOR_BLUE}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot - with alert icon', () => {
    const { toJSON } = render(
      <CustomAlert
        displayAlert
        alertIcon={CUSTOM_ALERT_MOCK.icon}
        alertTitleText={CUSTOM_ALERT_MOCK.title}
        alertMessageText={CUSTOM_ALERT_MOCK.message}
        positiveButtonText={CUSTOM_ALERT_MOCK.positiveButton}
        negativeButtonText={CUSTOM_ALERT_MOCK.negativeButton}
        onPositiveButtonPress={() => {}}
        onNegativeButtonPress={() => {}}
        positiveButtonBackgroundColor={ALERT_BUTTON_BACKGROUND_COLOR_BLUE}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('matches snapshot - with progress bar', () => {
    const { toJSON } = render(
      <CustomAlert
        displayAlert
        alertTitleText={CUSTOM_ALERT_MOCK.title}
        progress={1}
        alertMessageText={CUSTOM_ALERT_MOCK.message}
        negativeButtonText={CUSTOM_ALERT_MOCK.negativeButton}
        onNegativeButtonPress={() => {}}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
