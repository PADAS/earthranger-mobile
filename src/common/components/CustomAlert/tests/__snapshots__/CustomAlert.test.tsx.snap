// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomAlert matches snapshot - with alert icon 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000088",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "paddingBottom": 22,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 22,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="8"
            bbWidth="8"
            fill="#888B8D"
            focusable={false}
            height="8"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 8,
                  "width": 8,
                },
              ]
            }
            vbHeight={8}
            vbWidth={8}
            width="8"
          >
            <RNSVGGroup
              fill={4287138701}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M8 0L0 3.34667V3.78222L3.04 4.96L4.21333 8H4.64889L8 0Z"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          color="#000000"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#20303C",
                "textAlign": "left",
              },
              undefined,
              Object {
                "color": "#000000",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
                "fontSize": 18,
                "fontWeight": "500",
                "lineHeight": 21,
              },
            ]
          }
        >
          Alert Title
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#20303C",
              "textAlign": "left",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "color": "#63666A",
              "fontSize": 15,
              "fontWeight": "400",
              "marginTop": 16,
            },
          ]
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam semper in dolor et egestas.
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 22,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#888B8D",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "lineHeight": 18,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            Reject
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#0056C7",
              },
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 24,
                "maxWidth": "60%",
                "minWidth": "40%",
                "paddingHorizontal": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "lineHeight": 21,
                  "marginLeft": 9,
                  "marginRight": 9,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            Accept
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`CustomAlert matches snapshot - with progress bar 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000088",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "paddingBottom": 22,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 22,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <Text
          color="#000000"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#20303C",
                "textAlign": "left",
              },
              undefined,
              Object {
                "color": "#000000",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
                "fontSize": 18,
                "fontWeight": "500",
                "lineHeight": 21,
              },
            ]
          }
        >
          Alert Title
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#20303C",
              "textAlign": "left",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "color": "#63666A",
              "fontSize": 15,
              "fontWeight": "400",
              "marginTop": 16,
            },
          ]
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam semper in dolor et egestas.
      </Text>
      <View
        accessibilityRole="progressbar"
        accessibilityState={
          Object {
            "busy": true,
          }
        }
        accessibilityValue={
          Object {
            "max": 100,
            "min": 0,
            "now": 100,
          }
        }
        accessible={true}
        onLayout={[Function]}
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "#DDDDDD",
              "borderRadius": 5,
              "height": 12,
              "marginTop": 16,
              "opacity": 0,
              "overflow": "hidden",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 22,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#888B8D",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "lineHeight": 18,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            Reject
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`CustomAlert matches snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#00000088",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 5,
          "paddingBottom": 22,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 22,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "width": "100%",
          }
        }
      >
        <Text
          color="#000000"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#20303C",
                "textAlign": "left",
              },
              undefined,
              Object {
                "color": "#000000",
              },
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Object {
                "flex": 1,
                "fontSize": 18,
                "fontWeight": "500",
                "lineHeight": 21,
              },
            ]
          }
        >
          Alert Title
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#20303C",
              "textAlign": "left",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Object {
              "color": "#63666A",
              "fontSize": 15,
              "fontWeight": "400",
              "marginTop": 16,
            },
          ]
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam semper in dolor et egestas.
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 22,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#888B8D",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "lineHeight": 18,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            Reject
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "#0056C7",
              },
              Object {
                "alignItems": "center",
                "borderRadius": 5,
                "display": "flex",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 24,
                "maxWidth": "60%",
                "minWidth": "40%",
                "paddingHorizontal": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#20303C",
                  "textAlign": "left",
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 17,
                  "fontWeight": "600",
                  "lineHeight": 21,
                  "marginLeft": 9,
                  "marginRight": 9,
                  "textTransform": "capitalize",
                },
              ]
            }
          >
            Accept
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
